src/Controls/src/Core/Application/Application.cs

```diff
@@ -101,6 +101,15 @@ public IAppLinks AppLinks
 
 				_singleWindowMainPage = value;
 
+				if (value is not null)
+				{
+					OnParentResourcesChanged(this.GetMergedResources());
+					OnParentResourcesChanged([new KeyValuePair<string, object>("__ApplicationTheme__", _lastAppTheme)]);
+					((IElementDefinition)this).AddResourcesChangedListener(value.OnParentResourcesChanged);
+				} else if (MainPage is not null)
+				{
+					((IElementDefinition)this).RemoveResourcesChangedListener(MainPage.OnParentResourcesChanged);
+				}
 				if (Windows.Count == 1)
 				{
 					Windows[0].Page = value;
```

Comment:
Is this not triggering the parenting? Maybe if there are no windows - so I wonder if this new logic should be in an else?